controller/src/main/java/io/pravega/controller/server/security/auth/StreamAuthParams.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.auth.AuthHandler;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A helper class used for processing auth params for streams.
+ */
+@Slf4j
+public class StreamAuthParams {
+
+    private static final AuthorizationResource AUTH_RESOURCE = new AuthorizationResourceImpl();
+
+    private final String scope;
+    private final String stream;
+    private final boolean isInternalWritesWithReadPermEnabled;
+
+    @VisibleForTesting
+    @Getter
+    private final boolean isInternalStream;
+
+    @VisibleForTesting
+    StreamAuthParams(@NonNull String scope, @NonNull String stream) {
+        this(scope, stream, true);
+    }
+
+    public StreamAuthParams(@NonNull String scope, @NonNull String stream, boolean isInternalWritesWithReadPermEnabled) {
+        this.scope = scope;
+        this.stream = stream;
+        this.isInternalWritesWithReadPermEnabled = isInternalWritesWithReadPermEnabled;
+        if (stream.startsWith("_")) {
```

Comment:
What does this special string mean? Again, consider using `NameUtils`. If your method is not there, please add it.

Reply:
Sure, I've modified it to use `NameUtils`. I wasn't aware of that class. :)